main.go

```diff
@@ -176,6 +176,34 @@ func init() {
 	}
 }
 
+//Defines the interface for removing orphaned artifacts from aborted scans
+type CleanTemp interface {
+	//Removes orphaned directories from sources like Git
+	CleanTempDir(ctx context.Context, dirName string, pid int) error
+	//Removes orphaned files/artifacts from sources like Artifactory
+	CleanTempFiles(ctx context.Context, fileName string, pid int) error
+}
+
+func CleanTempDir(ctx context.Context, dirName string, pid int) error {
+	tempDir := os.TempDir()
+	files, err := os.ReadDir(tempDir)
+	if err != nil {
+		return fmt.Errorf("Error reading temp dir: %w", err)
+	}
+
+	pidStr := strconv.Itoa(pid)
+
+	for _, file := range files {
+		if file.IsDir() && strings.Contains(file.Name(), dirName) && !strings.Contains(file.Name(), pidStr) {			dirPath := fmt.Sprintf("%s/%s", tempDir, file.Name())
+			if err := os.RemoveAll(dirPath); err != nil {
+				return fmt.Errorf("Error deleting temp directory: %s", dirPath)
+			}
+			ctx.Logger().V(1).Info("Deleted directory", dirPath)
```

Comment:
The logger we're using uses optional key/values at the end of an `Info` argument list. In other words, it's expecting:

```go
Info(message, key1, value1, key2, value2, ...)
```

Since it's a vararg, it's syntactically valid to call it like `Info(message, key)` like we have here, and it won't surface until runtime as a weird looking log.

These sorts of mistakes are easy to both make and miss, but I'm not sure how to prevent them. 😕